src/urql/atomWithMutation.ts

```diff
@@ -19,17 +19,33 @@ export function atomWithMutation<Data, Variables extends AnyVariables>(
   createQuery: (get: Getter) => TypedDocumentNode<Data, Variables> | string,
   getClient: (get: Getter) => Client = (get) => get(clientAtom)
 ) {
-  const operationResultAtom = atom<
-    OperationResult<Data, Variables> | Promise<OperationResult<Data, Variables>>
-  >(
-    new Promise<OperationResult<Data, Variables>>(() => {}) // infinite pending
-  )
+  type Result = OperationResult<Data, Variables>
+  const operationResultAtom = atom((get) => {
+    let resolve: ((result: Result) => void) | null = null
+    const makePending = () => 
+      new Promise<Result>((r) => {
+        resolve = r
+      })
+
+    const resultAtom = atom<Result | Promise<Result>>(makePending())
+    return { resultAtom, makePending }
+  })
```

Comment:
On second look, this looks wrong. Do you plan to work on this refactor and fix?

Reply:
I wanted to fix infinite pending in operationResultAtom by calling the pending function when there's no result

Reply:
I'm also supposed to add typescript types to the values returned in operationResultAtom

Reply:
I thought this was a refactoring, but do you have any issues? In such a case, you want to reproduce the issue with test. That would  be very valuable.

Reply:
I will update test file and update this PR once all tests pass